os/arch/arm/src/amebasmart/amebasmart_idle.c

```diff
@@ -23,14 +23,181 @@
+					if (tizenrt_ready_to_sleep()) {
+#ifdef CONFIG_SMP
+						/*PG flow */
+						if (pmu_get_sleep_type() == SLEEP_PG) {
+							/* CPU1 just come back from pg, so can't sleep here */
+							if (pmu_get_secondary_cpu_state(1) == CPU1_WAKE_FROM_PG) {
+								goto EXIT;
+							}
+
+							/* CPU1 is in task schedular, tell CPU1 to enter hotplug */
+							if (pmu_get_secondary_cpu_state(1) == CPU1_RUNNING) {
+								/* CPU1 may in WFI idle state. Wake it up to enter hotplug itself */
+								up_irq_enable();
+								arm_gic_raise_softirq(1, 0);
+								arm_arch_timer_int_mask(0);
+								DelayUs(100);
+								goto EXIT;
+							}
+
+							/* For SMP case, timer should be set after confirming secondary core enter hotplug mode */
+							if (pmu_get_secondary_cpu_state(1) == CPU1_HOTPLUG) {
+								ap_timer_helper();
+							}
+							/* CG flow */
+						} else {
+							if (!check_wfi_state(1)) {
+								goto EXIT;
+							}
+						}
+#endif
+						/* Interrupt source from BT/UART will wake cpu up, just leave expected idle time as 0
+						Enter sleep mode for AP */
+						configPRE_SLEEP_PROCESSING(xModifiableIdleTime);
+						/* When wake from pg, arm timer has been reset, so a new compare value is necessary to
+						trigger an timer interrupt */
+						if (pmu_get_sleep_type() == SLEEP_PG) {
+							up_timer_enable();
+							arm_arch_timer_set_compare(arm_arch_timer_count() + 50000);
+						}
+						arm_arch_timer_int_mask(0);
+						configPOST_SLEEP_PROCESSING(xModifiableIdleTime);
+					}
+					else {
+						/* power saving when idle*/
+						arm_arch_timer_int_mask(0);
+						__asm(" DSB");
+						__asm(" WFI");
+						__asm(" ISB");
+					}
+#ifdef CONFIG_SMP
+EXIT:
+#endif				
+					/* Re-enable interrupts and sys tick*/
+					up_irq_enable();
+				}
+				/* If secondary core idle loop enters here, it should guarantee it has entered hotplug state*/
+				else if (up_cpu_index() == 1) {
+					if (pmu_get_sleep_type() == SLEEP_PG) {
+						if (tizenrt_ready_to_sleep()) {
+							/* CPU1 will enter hotplug state. Raise a task yield to migrate its task */
+							pmu_set_secondary_cpu_state(1, CPU1_HOTPLUG);
+						}
+					}
+
+					flags = irqsave();
+					__asm("	DSB");
+					__asm("	WFI");
+					__asm("	ISB");
+					up_irq_enable();
+#ifdef CONFIG_SMP
+					goto EXIT2;
+#endif
+				}
+				/* Wakeup from sleep, change the state back to PM_NORMAL */
+				ret = pm_changestate(PM_IDLE_DOMAIN, PM_NORMAL);
+				if (ret < 0) {
+					/* TODO: If we are unable to reset the state back to PM_NORMAL then what to do? */
+					pmdbg("State change failed after core waking up, please check logic flow!");
+					oldstate = PM_NORMAL;
+				}
+
+/* TODO: This exit is for secondary core
+   After 2nd core entered hotplug mode, TizenRT should remove the idle task for 2nd core
+   Revisit here to see if anything else need to be done */
+#ifdef CONFIG_SMP
+EXIT2:
+#endif
+/* If state transition is rejected, exit directly*/
+REJECTED:
```

Comment:
I think this REJECTED label should be outside the switch case, at the end of the up_idlepm function